extensions-contrib/statsd-emitter/src/main/java/org/apache/druid/emitter/statsd/StatsDEmitterConfig.java

```diff
@@ -61,7 +63,8 @@ public StatsDEmitterConfig(
       @JsonProperty("dimensionMapPath") String dimensionMapPath,
       @JsonProperty("blankHolder") String blankHolder,
       @JsonProperty("dogstatsd") Boolean dogstatsd,
-      @JsonProperty("dogstatsdConstantTags") List<String> dogstatsdConstantTags
+      @JsonProperty("dogstatsdConstantTags") List<String> dogstatsdConstantTags,
+      @JsonProperty("dogstatsdServiceAsTag") Boolean dogstatsdServiceAsTag
```

Comment:
nit: we are trying to get better about marking things that could be `@Nullable` as such, if you don't mind could you make this change to (I think) all the input parameters except `hostname` and `port`:
```java
@JsonProperty("prefix") @Nullable String prefix,
...
@JsonProperty("dogstatsdServiceAsTag") @Nullable Boolean dogstatsdServiceAsTag
```

It looks like all fields are given a default value _except_ for `dimensionMapPath`, so could you also [mark its field](https://github.com/apache/incubator-druid/pull/8238/files#diff-5af37bd75081c6bd11e9eae941a9d22cR46) and [its getter](https://github.com/apache/incubator-druid/pull/8238/files#diff-5af37bd75081c6bd11e9eae941a9d22cR185) as `@Nullable`? 

Side note, it sort of looks like most of the `null` checks in `equals` method are unnecessary because of the default values, but also probably harmless.

Reply:
Sure good idea to improve things by using `@Nullable`.

I made a draft change here: https://github.com/AlexandreYang/incubator-druid/pull/2/files

1) Added `@Nullable` to [all optional argument on the constructor](https://github.com/AlexandreYang/incubator-druid/pull/2/files#diff-5af37bd75081c6bd11e9eae941a9d22cR62-R69)

2) Adde `@Nullable` only to fields and getters that do do not have a default value: `dimensionMapPath`. Which seems to make sense since that's the only field that might be null if the getter `getDimensionMapPath()` is called. Similar approach taken here: [StorageLocationConfig.java](https://github.com/apache/incubator-druid/blob/c7eb7cd01837c48914ba284d08b6096b47c957b0/server/src/main/java/org/apache/druid/segment/loading/StorageLocationConfig.java).

Let me know if I'm missing something, otherwise, I can apply the changes from that draft change.

Reply:
Changes applied to this PR.